### PR TITLE
fix(compiler-cli): show a more specific error for Ivy NgModules

### PR DESCRIPTION
**Pull request by alxhub**
When an Ivy NgModule is imported into a View Engine build, it doesn't have
metadata.json files that describe it as an NgModule, so it appears to VE
builds as a plain, undecorated class. The error message shown in this
situation generic and confusing, since it recommends adding an @NgModule
annotation to a class from a library.

This commit adds special detection into the View Engine compiler to give a
more specific error message when an Ivy NgModule is imported.

### CODE REVIEW

**Review by AndrewKushnir — APPROVED** [review]

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Review by petebacondarwin — APPROVED** [review]
Reviewed-for: public-api

File: packages/compiler-cli/src/transformers/program.ts [hunk 1]
```diff
@@ -1058,6 +1058,33 @@ function syntaxErrorToDiagnostics(error: Error): Diagnostic[] {
...
+      return [{
+        messageText: `The NgModule '${ngModuleErrorData.className}' in '${
+            ngModuleErrorData
+                .fileName}' is imported by this compilation, but appears to be part of a library compiled for Angular Ivy. This may occur because:
+
+  1) the library was processed with 'ngcc'. Removing and reinstalling node_modules may fix this problem.
+
+  2) the library was published for Angular Ivy and v12+ applications only. Check its peer dependencies carefully and ensure that you're using a compatible version of Angular.
+
+See https://angular.io/errors/NG6999 for more information.
```
Comment (atscott) [inline packages/compiler-cli/src/transformers/program.ts:1079]: Is there a corresponding PR that adds this page?

Comment (alxhub) [reply packages/compiler-cli/src/transformers/program.ts:1079]: Not yet - I reached out to devrel about what content we want to add here.

**Review by jessicajaniuk — APPROVED** [review]
LGTM 🍪 

reviewed-for: public-api

**Review by zarend — APPROVED** [review]

**Pull request closed without merging**